### PR TITLE
Remove ChannelOutboundBuffer from the Unsafe interface

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

ChannelOutboundBuffer is an implementation detail of AbstractChannel, let's not expose it in our public interfaces. Beside this it also allowed users to mess with internals that could have "strange" side-effects. Beside this as ChannelOutboundBuffer had a package-private constructor it was also not really possible to write your own Channel implementation without extending AbstractChannel

Modifications:

Remove ChannelOutboundBuffer from the Channel.Unsafe interface

Result:

Don't expose implementations details

### CODE REVIEW

File: handler/src/main/java/io/netty5/handler/ssl/SslHandler.java [hunk 1]
```diff
@@ -1790,9 +1789,7 @@ public void handlerAdded0(final ChannelHandlerContext ctx) throws Exception {
             startHandshakeProcessing(active);
             // If we weren't able to include client_hello in the TCP SYN (e.g. no token, disabled at the OS) we have to
             // flush pending data in the outbound buffer later in channelActive().
-            final ChannelOutboundBuffer outboundBuffer;
-            if (fastOpen && ((outboundBuffer = channel.unsafe().outboundBuffer()) == null ||
-                    outboundBuffer.totalPendingWriteBytes() > 0)) {
+            if (fastOpen) {
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty5/handler/ssl/SslHandler.java:1792]: @chrisvest I think this is fine as `flush` is a no-op if there is nothing in the outbound buffer.

File: handler/src/main/java/io/netty5/handler/traffic/AbstractTrafficShapingHandler.java [hunk 2]
```diff
@@ -588,11 +588,14 @@ public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
         ctx.fireChannelRegistered();
     }
 
+    // TODO: Fix me later!
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty5/handler/traffic/AbstractTrafficShapingHandler.java:591]: @chrisvest same comment... I think we need to think about a better way how to support something like this. Maybe sending an event to let the transport know about this.

File: handler/src/test/java/io/netty5/handler/traffic/TrafficShapingHandlerTest.java [hunk 3]
```diff
@@ -33,13 +33,15 @@
...
 import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
 import java.util.concurrent.TimeUnit;
 
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertNull;
 
+@Disabled
```
Comment (normanmaurer) [inline handler/src/test/java/io/netty5/handler/traffic/TrafficShapingHandlerTest.java:44]: Disabled for now. Let's create an issue and fix it as a followup

File: handler/src/test/java/io/netty5/handler/traffic/FileRegionThrottleTest.java [hunk 4]
```diff
@@ -50,6 +50,7 @@
 
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+@Disabled
```
Comment (normanmaurer) [inline handler/src/test/java/io/netty5/handler/traffic/FileRegionThrottleTest.java:53]: Disabled for now. Let's create an issue and fix it as a followup

File: handler/src/main/java/io/netty5/handler/timeout/IdleStateHandler.java [hunk 5]
```diff
@@ -385,75 +360,6 @@ protected IdleStateEvent newIdleStateEvent(IdleState state, boolean first) {
         }
     }
 
-    /**
-     * @see #hasOutputChanged(ChannelHandlerContext, boolean)
-     */
-    private void initOutputChanged(ChannelHandlerContext ctx) {
```
[the remark this answers is not in the document]
Comment (chrisvest) [reply handler/src/main/java/io/netty5/handler/timeout/IdleStateHandler.java:391]: Yeah, if we need that, there ought to be a better way to do it.

File: handler/src/main/java/io/netty5/handler/traffic/AbstractTrafficShapingHandler.java (hunk 2 above)
Comment (chrisvest) [reply handler/src/main/java/io/netty5/handler/traffic/AbstractTrafficShapingHandler.java:591]: It seems niche enough that an event sounds like the right approach.

**Review by chrisvest — APPROVED** [review]

**Pull request merged**